src/yunohost/backup.py

```diff
@@ -49,111 +51,199 @@
+
+        if not os.path.isdir(self.collect_dir):
+            filesystem.mkdir(self.collect_dir, 0750, parents=True, uid='admin')
+        elif self.bindable:
             logger.debug("temporary directory for backup '%s' already exists",
-                tmp_dir)
-            if not _clean_tmp_dir():
+                         self.collect_dir)
+            if not self.clean():
                 raise MoulinetteError(
                     errno.EIO, m18n.n('backup_output_directory_not_empty'))
-        filesystem.mkdir(tmp_dir, 0750, parents=True, uid='admin')
 
-    # Initialize backup info
-    info = {
-        'description': description or '',
-        'created_at': timestamp,
-        'apps': {},
-        'hooks': {},
-    }
+    def _open_csv(self):
+        """ Initialize backup list """
+        self.csv_path = os.path.join(self.collect_dir, 'backup.csv')
+        try:
+            self.csv_file = open(self.csv_path, 'a')
+            self.fieldnames = ['source', 'dest']
+            self.csv = csv.DictWriter(self.csv_file, fieldnames=self.fieldnames,
+                                      quoting=csv.QUOTE_ALL)
+        except (IOError, OSError, csv.Error):
+            logger.error(m18n.n('backup_csv_creation_failed'))
+
+    def _get_env_var(self):
+        """ Define environment variable for hooks call """
+        env_var = {}
+
+        _, tmp_csv = tempfile.mkstemp(prefix='backupcsv_')
+        env_var['YNH_BACKUP_DIR'] = self.collect_dir
+        env_var['YNH_BACKUP_CSV'] = tmp_csv
+        return env_var
+
+    def _commit_path_in_csv(self, tmp_csv):
+        """ Commit collected path from system or app hooks """
+        with open(tmp_csv, 'r') as input:
+            while True:
+                data = input.read(100000)
+                if data == '':  # end of file reached
+                    break
+                self.csv_file.write(data)
+
+    def _mark_for_backup(self, source, dest=None):
+        """
+        Mark file or directory to backup
+
+        source: source path to backup
+        dest: destination path in the archive. If dest end by a slash the
+        basename of source is added
+
+        usage:
+        self._mark_for_backup('/var/www/wordpress', 'sources')
+        => wordpress dir will be move and rename in sources
+
+        self._mark_for_backup('/var/www/wordpress', 'sources/')
+        => wordpress dir will be put inside sources/ dir and won't be renamed
+
+        """
+        try:
+            if dest is None:
+                dest = source
+                source = os.path.join(self.collect_dir, source)
+            if dest.endswith("/"):
+                dest = os.path.join(dest, os.path.basename(source))
+            self.csv.writerow({'source': source, 'dest': dest})
+        except csv.Error:
+            logger.error(m18n.n('backup_csv_addition_failed'))
+
+    def _collect_hooks_files(self, hooks=[]):
+        """
+        Prepare backup for each selected system part
```

Comment:
Almost. It concerns hooks in data/hooks/backup/ (system hooks) and in /etc/yunohost/hooks.d/backup/ (user hooks)